src/mot/tracker/tracker.py

```diff
@@ -1,47 +1,169 @@
+        self.video_id = video_id
+
+        self.list_path_images = list_path_images
+        self.list_inference_output = list_inference_output
+
+        self.num_images = len(list_path_images)
+        self.list_geoloc = list_geoloc
+        self.fps = fps
+
+        self.detected_trash = []
+        self.tracking_done = False
+
+        self.iou_threshold = 0.3
+        self.rewind_window_match = 2
+
+    def potential_matching_trash_list(self, frame_idx, objects_per_frame_list):
+        '''Creates a list of trash which appear in the last `self.rewind_window_match` frames
+
+        Arguments:
+
+        - frame_idx: the frame index as integer
+        - objects_per_frame_list: the list of trash object for each past frame_idx
+
+        Returns:
+
+        - A list of trash objects that could potentially match
+        '''
+        potential_matching_ids = set()
+        for idx_rewind in range(frame_idx -1, frame_idx - self.rewind_window_match - 1, -1):
+            if idx_rewind >= 0:
+                potential_matching_ids = potential_matching_ids.union(set(objects_per_frame_list[idx_rewind]))
+        return [trash for trash in self.detected_trash if trash.id in potential_matching_ids]
+
+    def track_objects(self):
+        '''Main function which tracks trash objects. Assumes
+
+        Returns:
+
+        - The detected trash list
+        '''
+        if not self.list_inference_output:
+            raise ValueError("No inference was run, can't track objects")
+
+        nb_detected_objects = 0
+        objects_per_frame_list = [[] for f in self.list_inference_output]
+
+        for frame_idx, json_object in enumerate(self.list_inference_output):
+            found_classes = json_object.get("output/labels:0")
+            found_boxes = json_object.get("output/boxes:0")
+
+            # Build the list of previous trash that could be matched
+            potential_matching_trash = self.potential_matching_trash_list(frame_idx, objects_per_frame_list)
+            for label, box in zip(found_classes, found_boxes):
+                current_trash = Trash(nb_detected_objects, label, box, frame_idx)
+
+                # Is this object already matching something?
+                matching_id = current_trash.find_best_match_in_list(potential_matching_trash, self.iou_threshold)
+                if matching_id is not None:
+                    # append the frame & box to the matching trash
+                    self.detected_trash[matching_id].add_matching_object(box, frame_idx)
+                    objects_per_frame_list[frame_idx].append(matching_id)
+                else:
+                    # Otherwise, create new object
+                    self.detected_trash.append(copy.deepcopy(current_trash))
+                    objects_per_frame_list[frame_idx].append(current_trash.id)
+                    nb_detected_objects += 1
+
+        self.tracking_done = True
+        return self.detected_trash
```

Comment:
returning an attribute from a function is usualy a bad pattern. It means that the scoping of your attribute is too big.